Client/multiplayer_sa/CMultiplayerSA_WidescreenFix.cpp

```diff
@@ -0,0 +1,1111 @@
+    fHUDHeight[15] /= 1.125f;
+    fHUDHeight[16] /= 1.125f;
+}
+
+void UpdateScreenAspectRatio()
+{
+    fWideScreenWidthScale = 640.0f / (*(pfScreenAspectRatio) * 448.0f);
+    fWideScreenHeightScale = 448.0 / 448.0f;
+    fWideScreenWidthProperScale = static_cast<float>(pRsGlobal->width) / (*pfScreenAspectRatio * 448.0f);
+    fWideScreenHeightProperScale = static_cast<float>(pRsGlobal->height) / 448.0f;
+    fDefaultWidth = (4.0f / 3.0f);
+
+    UpdateFrontendFixes();
+    UpdateMiscFixes();
+    UpdateHUDFixes();
+    UpdateScriptFixes();
+}
+
+void InstallAspectRatioFixes()
+{
+    MemPut<BYTE>(0x745BD1, 0x7D);
+    MemPut<BYTE>(0x745BD9, 0x7C);
+
+    // Proportional coronas
+    MemSet((void*)0x6FB2C9, 0x90, 4);
+
+    MemPut<BYTE>(0x6FB2BD, 0x6C);
+    MemPut<BYTE>(0x6FB2DC, 0x78);
+    MemPut<BYTE>(0x713BE5, 0x20);
+    MemPut<BYTE>(0x713B6D, 0x38);
+    MemPut<BYTE>(0x713CFB, 0x38);
+    MemPut<BYTE>(0x713EFC, 0x30);
+    MemPut<BYTE>(0x714004, 0x38);
+
+    HookInstall(0x6FF420, (DWORD)CalculateAspectRatio, 5);
+
+    HookInstallCall(0x0053D695 - 0x1, (DWORD)UpdateScreenAspectRatio);
+    HookInstallCall(0x0053D7B2 - 0x1, (DWORD)UpdateScreenAspectRatio);
+    HookInstallCall(0x0053D967 - 0x1, (DWORD)UpdateScreenAspectRatio);
+    HookInstallCall(0x0053E771 - 0x1, (DWORD)UpdateScreenAspectRatio);
+    HookInstallCall(0x0053EB1A - 0x1, (DWORD)UpdateScreenAspectRatio);
+}
+
+void GetMemoryAddresses()
+{
+    pRsGlobal = (RsGlobal*)0xC17040;
+    pfScreenAspectRatio = (float*)0xC3EFA4;
+}
+
+float __stdcall StretchXHook(float fValue)
+{
+    if (pRsGlobal->width == 640)
+        return fValue;
+    else
+        return (float)pRsGlobal->width * fValue * 0.0015625f;
+}
+
+float __stdcall StretchYHook(float fValue)
+{
+    if (pRsGlobal->height == 448)
+        return fValue;
+    else
+        return (float)pRsGlobal->height * fValue * 0.002232143f;
+}
+
+void InstallFrontendFixes()
+{
+    // Font Scales
+    int m_dwFrontendWidth[] = {
+        0x5795CF,            // 0 Menu text
+        0x579981,            // 1 Menu text
+        0x579A01,            // 2 Menu text
+        0x57A36E,            // 3 Menu text
+        0x57A319,            // 4 Menu text
+        0x57A2A0,            // 5 Menu text
+        0x579718,            // 6 Menu text
+        0x5763C0,            // 7 Menu text
+        0x5749A5,            // 8 Menu text
+        0x57500F,            // 9 Menu text
+        0x579665,            // 10 Menu text
+        0x579863,            // 11 Menu text
+        0x57A1E3,            // 12 Menu text
+        0x57A3EB,            // 13 Menu text
+        0x582DDB,            // 14 Legend
+        0x582E0D,            // 15 Legend
+        0x582EB0,            // 16 Legend
+        0x583019,            // 17 Legend
+        0x58309C,            // 18 Legend
+        0x582F82,            // 19 Legend
+        0x583128,            // 20 Legend
+        0x5831AD,            // 21 Legend
+        0x57613B,            // 22 Legend
```

Comment:
I don't think we need to apply the fix for menu, legend, subtitles, wasted, busted etc

Reply:
removed